### PR TITLE
7.x.validator

### PR DESCRIPTION
**Pull request by MorganDawe**

### CODE REVIEW

**Comment by willtp87** [comment]
Bad root!

File: README.md [hunk 1]
```diff
@@ -30,3 +30,27 @@ CWRC-Writer API
...
+{base_url}:8080/validator/index.html
+* If not already configured, Add a reverse proxy setting on the server. 
+ex: edit sites-available/default add the following lines above the </VirtualHost> tag
+  ProxyPass /validator/ http://localhost:8080/validator/
+  ProxyPassReverse /validator/ http://localhost:8080/validator/
+* Restart apache. The validator should now be available to test at
+{base_url}/validator/index.html
+
+PLEASE NOTE: Any TEI-RDF data generated prior to this version of the cwrc-writer may not pass validation.
+Please regenerate the CWRC datastream to pass validation, if required.
```
Comment (willtp87) [inline README.md]: New line for git?

File: includes/admin.form.inc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Submit handler for the admin form.
+ *
+ * @param array $form
+ *   The admin form.
+ *
+ * @param array $form_state
+ *   The admin form state.
+ */
+function islandora_critical_edition_admin_submit($form, &$form_state) {
```
Comment (willtp87) [inline includes/admin.form.inc]: I would advise using https://api.drupal.org/api/drupal/modules!system!system.module/function/system_settings_form/7 and not using a dynamic default for the variable.

File: islandora_critical_edition.install [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+/**
+ * @file
+ * islandora_critical_edition.install
```
Comment (willtp87) [inline islandora_critical_edition.install]: Perhaps the name of the file may not be the right doc comment?

File: README.md (hunk 1 above)
Comment (MorganDawe) [inline README.md]: added new line.

File: theme/theme.inc [hunk 4]
```diff
@@ -26,6 +26,7 @@ function islandora_critical_edition_preprocess_islandora_critical_edition(&$vari
   $image_annotation_module_path = drupal_get_path('module', 'islandora_image_annotation');
   $pages = islandora_paged_content_get_pages($variables['islandora_object']);
   $pids = array_keys($pages);
+  $validate_path = variable_get('islandora_critical_edition_validate_url', $base_url . 'validator/validate.html');
```
Comment (willtp87) [inline theme/theme.inc]: Inconsistent default value.

File: includes/admin.form.inc [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * @return array
+ *   Islandora critical edition admin form.
+ */
+function islandora_critical_edition_admin($form, &$form_state) {
+  $form['islandora_critical_edition_validate_url'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Validation endpoint'),
+    '#default_value' => variable_get('islandora_critical_edition_validate_url', 'validator/validate.html'),
+    '#description' => t('Configure the cwrc-validator endpoint. This could point to an external URL.'),
```
Comment (willtp87) [inline includes/admin.form.inc]: I don't believe this can work if it points to an external URL.

File: README.md [hunk 6]
```diff
@@ -30,3 +30,27 @@ CWRC-Writer API
...
+you install the validator:
+* Build the .war file from source, or use the included .war file in the cwrc-validator repo
+(found here: https://github.com/cwrc/cwrc-validator.git, /target/cwrcxmlval-0.0.1-SNAPSHOT.war).
+* Rename the .war file to 'validator.war'.
+* Deploy the war to tomcats webapp directory (ex: ../tomcat/webapps).
+NOTE: Do not use the webapp manager to deploy this .war file.
+* After a moment, the .war should deploy and can be tested by visiting 
+{base_url}:8080/validator/index.html
+* If not already configured, Add a reverse proxy setting on the server. 
+ex: edit sites-available/default add the following lines above the </VirtualHost> tag
```
Comment (willtp87) [inline README.md]: Note this was on debian

File: README.md [hunk 7]
```diff
@@ -30,3 +30,27 @@ CWRC-Writer API
...
+  ProxyPass /validator/ http://localhost:8080/validator/
+  ProxyPassReverse /validator/ http://localhost:8080/validator/
+* Restart apache. The validator should now be available to test at
+{base_url}/validator/index.html
+
+PLEASE NOTE: Any TEI-RDF data generated prior to this version of the cwrc-writer may not pass validation.
```
Comment (willtp87) [inline README.md]: We're dropping this.

**Pull request merged**